travis/build_info.py

```diff
@@ -0,0 +1,207 @@
+"""
+Gather Travis usage info in real-time using Travis REST APIs
+
+This only applies to travis instances that do not require
+authorization (e.g., travis-ci.org). Auth is a #TODO
+
+"""
+
+import argparse
+import logging
+import requests
+import sys
+
+
+logger = logging.getLogger(__name__)
+
+BASE_URL = 'https://api.travis-ci.org/'
+
+
+def get_repos(org):
+    """
+    Returns list of active repos in a given org.
+    """
+
+    repo_list = []
+    # Note: One call is needed if org has <= 100 active repos.
+    # If org has > 100, this code will need to be modified
+    #  to include 'offset' requests
+    req = requests.get(
+        BASE_URL + 'v3/owner/{org}/repos?active=true'.format(org=org)
+    )
+    if req.status_code != 200:
+        raise requests.HTTPError(req.status_code)
+    req_json = req.json()
+    try:
+        for i in req_json['repositories']:
+            repo_list.append(i['name'])
+    except KeyError:
+        raise KeyError("Cannot parse response")
+
+    return repo_list
+
+
+def get_active_builds(org, repo):
+    """
+    Returns list of active builds for a given repo slug
+    """
+
+    repo_slug = '{org}/{repo}'.format(org=org, repo=repo)
+    req = requests.get(
+        BASE_URL + 'repos/{repo_slug}/builds'.format(repo_slug=repo_slug)
+    )
+    build_list = req.json()
+    active_build_list = []
+    for build in build_list:
+        if build['state'] != 'finished':
+            active_build_list.append(build)
+    return active_build_list
+
+
+# @property
+# def build_id(build_item):
+#
+#     return build_item['id']
+#
+
+def get_jobs(build_id):
+    """
+    Get the jobs for a build
+    return: list of dicts
+    """
+
+    req = requests.get(
+        BASE_URL + 'v3/build/{build_id}/jobs'.format(build_id=build_id)
+    )
+    return req['jobs']
+
+
+# @property
+# def job_state(job):
+#     """
+#     possible states:
+#     * received
+#     * queued
+#     * created
+#     * is there a started?
+#     """
+#
+#     return job['state']
+
+
+def repo_active_build_count(builds):
+    """
+    Returns counts of total builds, and total
+    running builds for a given list of them
+
+    This method assumes it has received a list
+    of active builds.
+    """
+    build_count = 0
+    started_count = 0
+    for build in builds:
+        build_count += 1
+        if build['state'] == 'started':
+            started_count += 1
+
+    return build_count, started_count
+
+
+def repo_active_job_count(active_builds):
+    """
+    Returns count of active jobs associated with builds in a
+    given repo.
+    """
```

Comment:
Can you please document the possible values for build stage and job state, or add a URL to the API specs? 
I couldn't find what the stage values might be with a quick google search. 
For state values, it seem from [states.rb#L6](https://github.com/travis-ci/travis.rb/blob/dcc9f20535c811068c4ff9788ae9bd026a116351/lib/travis/client/states.rb#L6) that the values are: created queued received started passed failed errored canceled ready
For build stages I don't see those [in the API spec](https://docs.travis-ci.com/api#builds), maybe I'm looking in the wrong place?